stdlib/src/sys/intrinsics.mojo

```diff
@@ -1504,6 +1504,34 @@ struct _RegisterPackType[*a: AnyTrivialRegType]:
         return __mlir_op.`kgen.pack.extract`[index = i.value](self.storage)
 
 
+# ===----------------------------------------------------------------------=== #
+# expect
+# ===----------------------------------------------------------------------=== #
+
+
+@always_inline("nodebug")
+fn expect[T: AnyTrivialRegType, //, expected_val: T](val: T) -> T:
+    """Provides information about expected (the most probable) value of `val`,
+    which can be used by optimizers.
+
+    Constraints:
+        Only work with integer types.
+
+    Parameters:
+        T: The type of the input value.
+        expected_val: The expected value of `val`.
+
+    Args:
+        val: The input value.
+
+    Returns:
+        The input value.
+    """
+    return llvm_intrinsic["llvm.expect", T, has_side_effect=False](
+        val, expected_val
```

Comment:
The times I've built functions that take infer only params and try to use positional params i.e. `expect[True](some_expression)` it leads to it trying to set the infer only param as that value, so you end up having to use kwarg. So in order to avoid the kwarg being long, this seems a bit better to me
```suggestion
fn expect[T: AnyTrivialRegType, //, value: T](val: T) -> T:
    """Provides information about expected (the most probable) value of `val`,
    which can be used by optimizers.

    Constraints:
        Only work with integer types.

    Parameters:
        T: The type of the input value.
        value: The expected value of `val`.

    Args:
        val: The input value.

    Returns:
        The input value.
    """
    return llvm_intrinsic["llvm.expect", T, has_side_effect=False](
        val, value
```